### PR TITLE
Upgrade to rails 5

### PR DESCRIPTION
**Pull request by ikennaokpala**
[Trello card](https://trello.com/c/D1IyHkh5/621-migrate-to-rails-5-calculators)

## Description 

This PR holds commits that setups, upgrade to rails 5 and fix error/warning as a result of the upgrade.
- Preparatory: These commit help increase the rails version range for dependencies like actionpack
    - Bump bundler and pry-byebug versions
    - Bump sass-rails to latest
    - Bump rspec-rails to latest
- Bump rails version to 5.0.2: This is contains a combined set of changes from [railsdiff](http://railsdiff.org/4.2.7.1/5.0.2) and `bundle exec rake rails:update`
- Use alphagov/airbrake 4: Suppresses rails 5 deprecation warning from airbrake
- Replace before_filter with before_action in controller
- Fix deprecation warnings from controller test

## Deprecation warnings

- ~~Passing strings or symbols to the middleware builder is deprecated, this exists in [RackStripClientIp::Middleware](https://github.com/alext/rack_strip_client_ip/blob/master/lib/rack_strip_client_ip/railtie.rb#L4)~~.
    - ~~It isn't clear yet the full impact of this, for now a PR [#2](https://github.com/alext/rack_strip_client_ip/pull/2) has been raised to address this.~~
- ~~https://github.com/alext/rack_strip_client_ip/pull/2#issuecomment-302735914~~

- after_filter is deprecation warning, this is caused by [slimmer]( https://github.com/alphagov/slimmer/blob/master/lib/slimmer/template.rb#L14) still makes use of
after_filter. So this deprecation warning will exist until slimmer is upgraded to use rails 5's *_action.
- [There is an open issue for slimmer](https://github.com/alphagov/slimmer/issues/196)

## Testing

https://deploy.integration.publishing.service.gov.uk/job/Deploy_App/13260/
https://errbit.integration.publishing.service.gov.uk/apps/5361171a0da115f1e0009de2

### CODE REVIEW

File: bin/rails [hunk 1]
```diff
@@ -1,4 +0,0 @@
-#!/usr/bin/env ruby
```
Comment (chao-xian) [inline bin/rails]: I don't think this file is meant to be deleted. That is not what railsdiff suggests doing. Shall I re-instate it @pmanrubia? Asking you in absence of Ikenna for advice.

Comment (chao-xian) [inline bin/rails]: I've undone this deletion.

Comment (pmanrubia) [inline bin/rails]: Hi @chao-xian did you do the update via `rake app:update`? It might be a more safe way to do it.

Comment (chao-xian) [inline bin/rails]: @pmanrubia ok so `rake rails:update` doesn't touch that file at all. Railsdiff suggests the change: http://railsdiff.org/4.2.7.1/5.0.2 - should I re-instate the `__FILE__`? That might call into question the upgrade by following Railsdiff (except when this file was deleted, which I've put back and followed the Railsdiff change).

Comment (chao-xian) [inline bin/rails]: Spoken to @pmanrubia who gives the 👍  with how I've done this.

**Comment by nickcolley** [comment]
Being cheeky, do you think we could also update https://github.com/alphagov/calculators/blob/efe2a857fc2343763876bd36cd23d2262e8653de/spec/features/child_benefit_tax_calculator_spec.rb#L177 to use the allow_label_click method? 👼

**Comment by ikennaokpala** [comment]
@nickcolley will have a look at it

**Pull request merged**